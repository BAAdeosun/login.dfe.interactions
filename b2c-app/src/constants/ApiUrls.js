export const API_URLS = {
    //TODO replace this URL with placeholder __--changeEmailAPI--__ when env vars are ready
    CHANGE_EMAIL: "https://dev-api-customerengagement.platform.education.gov.uk/authncs/ChangeEmail"
};