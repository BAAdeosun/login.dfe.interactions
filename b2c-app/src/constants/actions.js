export const ACTIONS = {
    SIGNUP: 'signup',
    LOGIN: 'login',
    RESET_PASSWORD: 'reset-password',
    FIND_EMAIL: 'find-email',
    CHANGE_EMAIL: 'change-email'
};