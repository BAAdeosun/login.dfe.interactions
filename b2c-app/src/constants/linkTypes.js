export const LINK_TYPES = {
    LINK: 'link',
    BUTTON: 'button',
    API_CALL: 'api'
};