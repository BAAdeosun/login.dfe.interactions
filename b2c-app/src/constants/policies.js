export const POLICIES = {
    SIGNIN_INVITATION: 'B2C_1A_Signin_Invitation',
    ACCOUNT_SIGNUP: 'B2C_1A_Account_Signup',
    PASSWORD_RESET: 'B2C_1A_Password_Reset',
    PASSWORD_RESET_CONFIRMATION: 'B2C_1A_Password_Reset_Confirmation',
    FIND_EMAIL: 'B2C_1A_Find_Email',
    SIGNUP_CONFIRMATION: 'B2C_1A_Signup_Confirmation',
    SIGNUP_INVITATION: 'B2C_1A_Signup_Invitation',
    CHANGE_EMAIL: 'B2C_1A_Change_Email'
};