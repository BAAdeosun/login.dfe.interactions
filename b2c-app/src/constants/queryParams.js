export const QUERY_PARAMS = {
    CLIENT_ID: 'client_id',
    REDIRECT_URI: 'redirect_uri'
}