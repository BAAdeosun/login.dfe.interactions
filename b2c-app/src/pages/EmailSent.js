import React from 'react';
import components from '../components';

export default function EmailSent() {
    return (
        <div id="emailSent">
            <div className="govuk-width-container">
                <components.Breadcrumbs />

                <main className="govuk-main-wrapper">
                    <div className="govuk-grid-row">
                        <div className="govuk-grid-column-two-thirds">
                            <components.PageTitle size='xl' title="We've sent you an email"/>
                            <components.B2C />
                            <p className="govuk-body">Check your spam folder if you can't see it in a few minutes.</p>
                            {/* <p className="govuk-body">If you don't receive an email after this time you can resend the activation email.</p> */}
                            <p className="govuk-body">This link expires in 24 hours.</p>
                            <p className="govuk-body">
                                <a href="./signup" className="govuk-link">I entered the wrong email address</a>
                            </p>
                        </div>
                    </div>
                </main>

            </div>
        </div>
    )
}
