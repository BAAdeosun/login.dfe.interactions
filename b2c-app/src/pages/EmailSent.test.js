import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { POLICIES } from '../constants/policies';

import EmailSent from './EmailSent';

it('renders without crashing', () => {
    shallow(<EmailSent />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<EmailSent />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('when policy is POLICIES.SIGNUP_INVITATION', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.SIGNUP_INVITATION} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when policy is POLICIES.ACCOUNT_SIGNUP', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.ACCOUNT_SIGNUP} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when policy is POLICIES.SIGNUP_CONFIRMATION', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.SIGNUP_CONFIRMATION} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when policy is POLICIES.SIGNIN_INVITATION', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.SIGNIN_INVITATION} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when policy is POLICIES.CHANGE_EMAIL', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.CHANGE_EMAIL} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when policy is POLICIES.RESEND_EMAIL', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.RESEND_EMAIL} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when policy is POLICIES.PASSWORD_RESET', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.PASSWORD_RESET} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when policy is POLICIES.PASSWORD_RESET_CONFIRMATION', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<EmailSent policy={POLICIES.PASSWORD_RESET_CONFIRMATION} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});