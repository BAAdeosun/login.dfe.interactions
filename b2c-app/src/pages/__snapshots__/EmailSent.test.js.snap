// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.ACCOUNT_SIGNUP renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={null}
                >
                  resend the activation email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.CHANGE_EMAIL renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={[Function]}
                >
                  resend the activation email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
              <a
                className="govuk-button govuk-button--start"
                data-module="govuk-button"
                draggable="false"
                id="signInLink"
                onClick={null}
                role="button"
              >
                Return to sign in
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.PASSWORD_RESET renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                You will only receive this email if you have an account.
              </p>
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={null}
                >
                  resend password reset email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.PASSWORD_RESET_CONFIRMATION renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                You will only receive this email if you have an account.
              </p>
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={null}
                >
                  resend password reset email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.RESEND_EMAIL renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={[Function]}
                >
                  resend the activation email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
              <a
                className="govuk-button govuk-button--start"
                data-module="govuk-button"
                draggable="false"
                id="signInLink"
                onClick={null}
                role="button"
              >
                Return to sign in
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.SIGNIN_INVITATION renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={[Function]}
                >
                  resend the activation email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
              <p
                className="govuk-body"
              >
                If you changed your details to an incorrect email address and have been locked out of your account, 
                <a
                  className="govuk-link"
                  href="https://nationalcareers.service.gov.uk/contact-us"
                  id="contactUsLink"
                  onClick={null}
                >
                  contact us
                </a>
                .
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.SIGNUP_CONFIRMATION renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={null}
                >
                  resend the activation email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`when policy is POLICIES.SIGNUP_INVITATION renders correctly 1`] = `
<div
  id="emailSent"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've sent you an email
            </h1>
            <div>
              <p
                className="govuk-body"
                id="emailSentTo"
              />
              <p
                className="govuk-body"
              >
                You will only receive this email if you have an account.
              </p>
              <p
                className="govuk-body"
              >
                Check your spam folder if you cannot see it in a few minutes.
              </p>
              <p
                className="govuk-body"
              >
                If you do not receive an email after this time you can 
                <a
                  className="govuk-link"
                  id="resendEmailLink"
                  onClick={null}
                >
                  resend the activation email
                </a>
                .
              </p>
              <p
                className="govuk-body"
              >
                This link expires in 24 hours.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
